modules/library/render/src/test/java/org/geotools/renderer/lite/LetterConflictTest.java

```diff
@@ -0,0 +1,265 @@
+        StreamingRenderer renderer = new StreamingRenderer();
+        Map rendererParams = new HashMap();
+        LabelCacheImpl labelCache = new LabelCacheImpl();
+        rendererParams.put(StreamingRenderer.LABEL_CACHE_KEY, labelCache);
+        renderer.setRendererHints(rendererParams);
+        renderer.setJava2DHints(new RenderingHints(KEY_ANTIALIASING, VALUE_ANTIALIAS_ON));
+        renderer.setContext(context);
+        return renderer;
+    }
+
+    public void testLetterConflictEnabled() throws Exception {
+
+        LabelCacheImpl.DISABLE_LETTER_LEVEL_CONFLICT = true;
+        Style style = RendererBaseTest.loadStyle(this, "letterConflict20.sld");
+        DefaultMapContext mc = new DefaultMapContext(DefaultGeographicCRS.WGS84);
+        mc.addLayer(fs_line1, style);
+        StreamingRenderer renderer = getNewRenderer(mc);
+        final BufferedImage image1 = RendererBaseTest.renderImage(renderer, bounds1, null);
+
+        LabelCacheImpl.DISABLE_LETTER_LEVEL_CONFLICT = false;
+        style = RendererBaseTest.loadStyle(this, "letterConflict20.sld");
+        mc = new DefaultMapContext(DefaultGeographicCRS.WGS84);
+        mc.addLayer(fs_line1, style);
+        renderer = getNewRenderer(mc);
+        final BufferedImage image2 = RendererBaseTest.renderImage(renderer, bounds1, null);
+
+        assertTrue("More labels in image2 than image1",
+                countPixels(image2, Color.BLACK) > countPixels(image1, Color.BLACK));
+
+        // // Write to file
+        // File tmpFile = File.createTempFile("geotools",".png");
+        // ImageIO.write(image1, "png", tmpFile);
+        // tmpFile = File.createTempFile("geotools",".png");
+        // ImageIO.write(image2, "png", tmpFile);
+        // // Interactive visualization
+        // showImage("letterConflictEnabled false", TIME, image1);
+        // showImage("letterConflictEnabled true", TIME, image2);
+    }
+
+    public void testLetterConflictEnabled2Lines() throws Exception {
+
+        LabelCacheImpl.DISABLE_LETTER_LEVEL_CONFLICT = true;
+        Style style = RendererBaseTest.loadStyle(this, "letterConflict20.sld");
+        DefaultMapContext mc = new DefaultMapContext(DefaultGeographicCRS.WGS84);
+        mc.addLayer(fs_line2, style);
+        StreamingRenderer renderer = getNewRenderer(mc);
+        final BufferedImage image1 = RendererBaseTest.renderImage(renderer, bounds1, null);
+
+        LabelCacheImpl.DISABLE_LETTER_LEVEL_CONFLICT = false;
+        style = RendererBaseTest.loadStyle(this, "letterConflict20.sld");
+        mc = new DefaultMapContext(DefaultGeographicCRS.WGS84);
+        mc.addLayer(fs_line2, style);
+        renderer = getNewRenderer(mc);
+        final BufferedImage image2 = RendererBaseTest.renderImage(renderer, bounds1, null);
+
+        assertTrue("More labels in image2 than image1",
+                countPixels(image2, Color.BLACK) > countPixels(image1, Color.BLACK));
+
+        // // Write to file
```

Comment:
You are doing a pretty good job of following our code formatting conventions - thanks :)

Can I ask why this commented out "dead code" is being included in the pull request?